### PR TITLE
Add first and last names to performance report

### PR DESCRIPTION
**Pull request by karenc**
- Remove GetStudentProfiles and CourseMembership::GetStudents

  No longer in use.

- Add `CourseMembership::Models::Period.active_enrollments` association

- Add first and last names to performance report

- Add first and last names to performance report exports

- Remove extra delegate in CourseMembership::Models::Student

### CODE REVIEW

File: app/routines/get_student_profiles.rb [hunk 1]
```diff
@@ -1,32 +1,19 @@
 class GetStudentProfiles
```
Comment (jpslav) [inline app/routines/get_student_profiles.rb]: This routine is now very similar to `GetStudentRoster`.  Could it maybe be used in place of this routine?  (would have to add a group_by period call). If so, we can ditch `GetStudentProfiles`.

File: app/subsystems/course_membership/models/period.rb [hunk 2]
```diff
@@ -16,7 +19,7 @@ class CourseMembership::Models::Period < Tutor::SubSystems::BaseModel
   default_scope { order(:name) }
 
   def student_roles
-    enrollments.latest.active.includes(student: :role).collect{ |en| en.student.role }
+    active_enrollments.includes(student: :role).collect{ |en| en.student.role }
```
Comment (jpslav) [inline app/subsystems/course_membership/models/period.rb]: for my own benefit, can you explain how this call works without a `period` being passed as indicated on line 11 above?

**Comment by jpslav** [comment]
Looks good.  One substantive comment to address and then a rebase.

File: app/routines/get_student_profiles.rb (hunk 1 above)
Comment (karenc) [inline app/routines/get_student_profiles.rb]: Student roster has period information, maybe we can just filter it?

File: app/subsystems/course_membership/models/period.rb (hunk 2 above)
Comment (karenc) [inline app/subsystems/course_membership/models/period.rb]: I think `has_many` with lambda has an optional argument which is the object itself (a `CourseMembership::Models::Period` instance) if I understand it correctly.

For example:
`CourseMembership::Models::Period.first.student_roles`, the `active_enrollments` `period` argument is `CourseMembership::Models::Period.first`.

Don't know if this answers your question.

File: app/routines/get_student_profiles.rb (hunk 1 above)
Comment (karenc) [inline app/routines/get_student_profiles.rb]: Oh turns out... after a rebase onto master, there's no code using `GetStudentProfiles` anymore. ;D  I'll remove it.

**Comment by jpslav** [comment]
This partially addresses #378

**Pull request merged**